lightning_pose/models/heatmap_tracker.py

```diff
@@ -269,16 +269,40 @@ def forward(
         # softmax temp stays 1 here; to modify for model predictions, see constructor
         return spatial_softmax2d(heatmaps, temperature=torch.tensor([1.0]))
 
+    def convert_bbox_coords(self, batch_dict: Union[HeatmapLabeledBatchDict, UnlabeledBatchDict],
+                            predicted_keypoints: TensorType["batch", "num_targets"]
+                            ) -> TensorType["batch", "num_targets"]:
+        # reshape from (batch, nTargets) back to (batch, nKey, 2), in x,y order
+        predicted_keypoints = predicted_keypoints.reshape((-1, self.num_keypoints, 2))
+        # divide by image dims to get 0-1 normalized coordinates
+        if "images" in batch_dict.keys():
+            predicted_keypoints[:, :, 0] /= batch_dict["images"].shape[-1]  # -1 dim is width "x"
+            predicted_keypoints[:, :, 1] /= batch_dict["images"].shape[-2]  # -2 dim is height "y"
+        else:  # we have unlabeled dict, 'frames' instead of 'images'
+            predicted_keypoints[:, :, 0] /= batch_dict["frames"].shape[-1]  # -1 dim is width "x"
+            predicted_keypoints[:, :, 1] /= batch_dict["frames"].shape[-2]  # -2 dim is height "y"
+        # multiply and add by bbox dims (x,y,h,w)
+        for i in range(predicted_keypoints.shape[0]):
+            predicted_keypoints[i, :, 0] *= batch_dict["bbox"][i, 3]  # scale x by box width
+            predicted_keypoints[i, :, 0] += batch_dict["bbox"][i, 0]  # add bbox x offset
+            predicted_keypoints[i, :, 1] *= batch_dict["bbox"][i, 2]  # scale y by box height
+            predicted_keypoints[i, :, 1] += batch_dict["bbox"][i, 1]  # add bbox y offset
```

Comment:
can this be vectorized? should check if something like `predicted_keypoints[:, :, 0] *= batch_dict["bbox"][:, 3]` works (might need `batch_dict["bbox"][:, 3].unsqueeze(1)` or something)

Reply:
unsqueeze works!